src/probnum/filtsmooth/gaussfiltsmooth/linearizing_transition.py

```diff
@@ -0,0 +1,70 @@
+"""Interface for approximate transitions.
+
+Applied to e.g. extended and unscented Kalman filtering and smoothing.
+"""
+
+import abc
+import typing
+
+import numpy as np
+
+import probnum.random_variables as pnrv
+import probnum.type as pntype
+from probnum.filtsmooth import statespace
+
+
+# Naming may be an overfit to EKF, but the next more general name that admits UKF
+# would be something like "ApproximateTransition" which says less than LinearizingTransition?
+class LinearizingTransition(statespace.Transition, abc.ABC):
```

Comment:
Is it not rather a LinearizedTransition?

Reply:
Alright, coming back here since I understand a little more now. I see now how this is not a LinearizedTransition. But overall I find this part a bit confusing.

Vaguely speaking, maybe there _should_ be something like a "linearized transition" somewhere, explicitly? Or have `linearize` have a return value, instead of changing some internal variable?

But this feels hard to discuss here and I'm also definitely not close enough to the code right now. Maybe we could just have a chat about that design decision tomorrow? :)

Reply:
removed it